### PR TITLE
clients/web: server-side redirection for logged-in/out users

### PR DESCRIPTION
**Pull request by frankie567**
The idea is to leverage Next.js server-side redirects to automatically redirect the user if they are logged-in or not.

The good thing is that it'll save us some "flashing" states when accessing login page while authenticated and vice versa.

Ultimately, this might allow us to remove the `Gatekeeper` component.

As a bonus, when accessing `polar.sh`, the user is automatically redirected to `polar.sh/feed` is they are logged in (not sure if desired, but I personally like the behavior. Can remove it otherwise).

---

To make this work, we simply check for the existence of our session cookie, `polar`, which I made a root domain cookie (see https://github.com/polarsource/polar/commit/11cac1d451cb935a87defe739241b69c3dfbbb65)

### CODE REVIEW

**Comment by frankie567** [comment]
Would gladly have your call on this, guys 😄

File: clients/apps/web/next.config.js
Comment (zegl) [inline clients/apps/web/next.config.js]: This is great! Really nice idea!

There is a few more pages that can handle logged out users:

* `/signup/maintainer`
* `/github/installation` (supports both logged in and logged out users)
* `/github/session`

and maybe even a few more? What about the magic link pages?

Comment (zegl) [reply clients/apps/web/next.config.js]: Also all `/api/`, `.well-known/`, `/embed/`, and `/og/` routes

Comment (zegl) [reply clients/apps/web/next.config.js]: `/new/`

Comment (zegl) [reply clients/apps/web/next.config.js]: And all public user pages, the pledge pages, etc.

Comment (zegl) [reply clients/apps/web/next.config.js]: It might actually be easier to add a list of pages here that are for logged in users only than doing the opposite. That way we don't need to list all pages in the route, as we would handle it client side of we miss to configure a page.

Comment (frankie567) [reply clients/apps/web/next.config.js]: Good point 😄 I didn't think we had so many public pages!

Comment (zegl) [reply clients/apps/web/next.config.js]: It was more than I thought too! :D

Comment (frankie567) [reply clients/apps/web/next.config.js]: Much cleaner like this indeed!

**Review by zegl — APPROVED** [review]
This looks very nice

**Comment by emilwidlund** [comment]
> This looks very nice

Indeed! Good job @frankie567 🥳

**Review by emilwidlund — APPROVED** [review]

**Pull request merged**
